src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,26 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private Engine engine;
+    private String model;
+    private int year;
+    private int avgSpeed;
+    private int maxSpeed;
+
+    public Car(Engine engine, String model, int year, int avgSpeed, int maxSpeed) {
+        this.engine = engine;
+        this.model = model;
+        this.year = year;
+        this.avgSpeed = avgSpeed;
+        this.maxSpeed = maxSpeed;
+    }
+
+    @Override
+    public Car clone() {
+        Engine clonedEngine = this.engine.clone();
```

Comment:
```suggestion
        Engine clonedEngine = engine.clone();
```
word `this` is redundant here and with other fields too